core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -458,10 +457,12 @@ class GroupCoordinator(val brokerId: Int,
   }
 
   private def addMemberAndRebalance(sessionTimeoutMs: Int,
+                                    clientId: String,
                                     protocols: List[(String, Array[Byte])],
                                     group: GroupMetadata,
                                     callback: JoinCallback) = {
-    val memberId = group.generateNextMemberId
+    // use the client-id with a random id suffix as the member-id
+    val memberId = clientId + "-" + group.generateMemberIdSuffix
```

Comment:
I think this is orthogonal to coordinator failover case: the client id is just used for easy debugging / tooling in addition to a global unique but random member id, it will just add a few bytes to the stored message but should be fine with Kafka.